src/authentication-strategies/JWT.strategy.ts

```diff
@@ -0,0 +1,30 @@
+const jwt = require('jsonwebtoken');
+import {promisify} from 'util';
+const verifyAsync = promisify(jwt.verify);
+// Consider turn it to a binding
+const SECRET = 'secretforjwt';
+import {Request, HttpErrors} from '@loopback/rest';
+import {UserProfile} from '@loopback/authentication';
+import * as _ from 'lodash';
+
+export class JWTStrategy {
+  // tslint:disable-next-line:no-any
+  async authenticate(request: Request): Promise<UserProfile | undefined> {
+    // there is a discussion regarding how to retrieve the token,
+    // see comment https://github.com/strongloop/loopback-next/issues/1997#issuecomment-451054806
+    const token = request.query.token || request.headers['authorization'];
+    if (token) {
+      try {
+        const decoded = await verifyAsync(token, SECRET);
+        return Promise.resolve(_.pick(decoded, ['id', 'email']));
```

Comment:
I think you should pick `name` too, since that's a property described in UserProfileSchema below?

Reply:
@bajtos The picked properties are a little bit tricky.

- The type `UserProfile` only has `id` as required, which makes sense. `email` and `name` are optional properties.
- The `User` model doesn't have a `name` property but has `firstName` and `lastName` instead, we could compose a `name` property by joining them, but given the fact that `name` is optional, is it necessary to compose one?

Reply:
This is tricky indeed.

I think we should make `UserProfile` more flexible to allow the applications to choose what fields they want to include. That's probably out of scope of this spike though.

I would assume that UserProfile's `name` property is meant either as a handle or a display name. If we have firstName and lastName properties in our domain model, then I propose to use `firstName` as the profile `name`.

@raymondfeng what do you think?

Reply:
> I would assume that UserProfile's name property is meant either as a handle or a display name. If we have firstName and lastName properties in our domain model, then I propose to use firstName as the profile name.

I am good with doing it in this PR, and for the discussion about the flexibility, I created a new issue for it, let's continue the discussion there: https://github.com/strongloop/loopback-next/issues/2246